cep-??.md

```diff
@@ -0,0 +1,246 @@
+<table>
+<tr><td> Title </td><td> The <code>MatchSpec</code> grammar </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Jaime Rodríguez-Guerra &lt;jaime.rogue@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> June 4, 2024 </td></tr>
+<tr><td> Updated </td><td> June 4, 2024 </td></tr>
+<tr><td> Discussion </td><td> https://github.com/conda/ceps/pull/82 </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+This CEP standardizes the grammar for the `MatchSpec` query language.
+
+## Motivation
+
+The motivation of this CEP is merely informative. It describes the details of an existing grammar.
+
+## Nomenclature
+
+The `MatchSpec` query syntax is a mini-language designed to query package records from one or more conda channels. It is sometimes referred to as simply _spec_.
+
+## Mini language
+
+The `MatchSpec` mini language has gone through several iterations.
+
+The simplest form merely consists of up to three positional arguments: `name [version [build]]`. Only `name` is required. `version` can be any version specifier. `build` can be any string matcher. See "Match conventions" below.
+
+The positional syntax also allows the `=` character as a separator, instead of a space. When this is the case, versions are interpreted differently. `pkg=1.8` will be taken as `1.8.*` (fuzzy), but `pkg 1.8` will give `1.8` (exact). To have fuzzy matches with the space syntax, you need to use `pkg =1.8`. This nuance does not apply if a `build` string is present; both `foo==1.0=*` and `foo=1.0=*` are equivalent (they both understand the version as `1.0`, exact).
+
+`MatchSpec` queries can be also specified with keyword arguments between square brackets. Key-value pairs can be delimited by comma, space, or comma+space. Value can optionally be wrapped in single or double quotes, but must be wrapped if `value` contains a comma, space, or equal sign. Their values can be quoted with single or double quotes. The accepted keys are:
+
+- `channel` (`str`): Name or URL of a channel
+- `subdir` (`str`): Identifier of the subdir (either platform-specific or `noarch`)
+- `version` (`str`): A version specifier
+- `build` (`str`): Build string or a glob match.
+- `build_number` (`int`): Number of the build
+- `md5` (`str`): MD5 hash of the artifact
+- `sha256` (`str`): SHA256 hash of the artifact
+
+These are also accepted but have reduced utility. Their usage is discouraged:
+
+- `url`
```

Comment:
Note that a full URL can be parsed into a MatchSpec object so... should a URL be considered a valid form? In those cases, a note: parsers need to account for %-decoding. See xref https://github.com/conda/conda/pull/14481.